dm/pkg/conn/utils.go

```diff
@@ -144,23 +141,69 @@ func GetMasterStatus(ctx *tcontext.Context, db *BaseDB, flavor string) (
 		| 0-1-2                    |
 		+--------------------------+
 	*/
+
+	var rowsResult [][]string
 	if flavor == gmysql.MySQLFlavor {
-		err = rows.Scan(&binlogName, &pos, &binlogDoDB, &binlogIgnoreDB, &gtidStr)
+		rowsResult, err = export.GetSpecifiedColumnValuesAndClose(rows, "File", "Position", "Binlog_Do_DB", "Binlog_Ignore_DB", "Executed_Gtid_Set")
+		if err != nil {
+			err = terror.DBErrorAdapt(err, terror.ErrDBDriverError)
+			return binlogName, pos, binlogDoDB, binlogIgnoreDB, gtidStr, err
+		}
+
+		switch {
+		case len(rowsResult) == 0:
+			err = terror.ErrNoMasterStatus.Generate()
+			return binlogName, pos, binlogDoDB, binlogIgnoreDB, gtidStr, err
+		case len(rowsResult[0]) != 5:
+			ctx.L().Warn("The number of columns that SHOW MASTER STATUS returns for MySQL is not equal to 5, will not use the retrieved information")
```

Comment:
```suggestion
			ctx.L().DPanic("The number of columns that SHOW MASTER STATUS returns for MySQL is not equal to 5, will not use the retrieved information")
```
and we should return an error for this case, like `err = terror.ErrNoMasterStatus.Generate()`. Otherwise caller may has no ability to handle that case